### PR TITLE
Fix test setup metric validation 

### PR DESCRIPTION
**Pull request by karlkfi**
- Always validate a standard set of metrics after test setup.
- Record the expected "declared resources" populated by test setup.
  This is used when validating standard metrics or as a base-line
  when calculating declared resources from a test.
- Clean up setupDelegatedControl and setupCentralizedControl to use
  RootRepos/NonRootRepos created by setupTestCase, rather than using
  the opts they came from.
- Configure reconcile timeout consistently.

### CODE REVIEW

File: e2e/nomostest/config_sync.go [hunk 1]
```diff
@@ -979,6 +973,24 @@ func setupCentralizedControl(nt *NT, opts *ntopts.New) {
 	}
 }
 
+func validateRootSyncsExist(nt *NT, opts *ntopts.New) error {
+	var err error
+	for rsName := range opts.RootRepos {
+		err = multierr.Append(err,
+			nt.Validate(rsName, configsync.ControllerNamespace, &v1beta1.RootSync{}))
+	}
+	return nil
```
Comment (sdowell) [inline e2e/nomostest/config_sync.go:982]: return `err`

ditto for function below

Comment (karlkfi) [reply e2e/nomostest/config_sync.go]: fixed

**Comment by karlkfi** [comment]
Significant changes. Needs another review.

File: e2e/nomostest/nt.go [hunk 2]
```diff
@@ -425,25 +437,16 @@ func (nt *NT) MustMergePatch(obj client.Object, patch string, opts ...client.Pat
...
-		numObjects += numRepoSyncs               // 1 for each RepoSync
-		numObjects += numRepoSyncs               // 1 for each RepoSync RoleBinding
-		if isPSPCluster() {
-			numObjects += numRepoSyncs // 1 for each RepoSync ClusterRoleBinding
-		}
-	}
-	return numObjects
+// DefaultRootSyncObjectCount returns the number of declared resource objects in
+// the specified RootSync, according to records from test setup.
+func (nt *NT) DefaultRootSyncObjectCount(syncName string) int {
```
Comment (sdowell) [inline e2e/nomostest/nt.go:442]: I think something like `InitialR*SyncObjectCount` might be a bit more clear than `DefaultR*SyncObjectCount`

File: e2e/nomostest/config_sync.go [hunk 3]
```diff
@@ -873,118 +855,121 @@ func RepoSyncObjectV1Beta1FromOtherRootRepo(nt *NT, nn types.NamespacedName, rep
...
-		if err := nt.WatchForAllSyncs(); err != nil {
-			nt.T.Fatal(err)
-		}
-		return
-	}
-
-	cr := nt.RepoSyncClusterRole()
-	nt.RootRepos[configsync.RootSyncName].Add("acme/cluster/cr.yaml", cr)
+	if len(nt.NonRootRepos) > 0 {
+		cr := nt.RepoSyncClusterRole()
```
Comment (sdowell) [inline e2e/nomostest/config_sync.go:879]: Why the change to only create the `RepoSyncClusterRole` if the ntopts define a RepoSync? There are some tests which create RepoSyncs within the test rather than via the test setup. This seems like it could be a gotcha

File: e2e/nomostest/config_sync.go [hunk 4]
```diff
@@ -873,118 +855,121 @@ func RepoSyncObjectV1Beta1FromOtherRootRepo(nt *NT, nn types.NamespacedName, rep
...
-		// Wait for the RootSyncs and exit early
-		if err := nt.WatchForAllSyncs(); err != nil {
-			nt.T.Fatal(err)
-		}
-		return
-	}
-
-	cr := nt.RepoSyncClusterRole()
-	nt.RootRepos[configsync.RootSyncName].Add("acme/cluster/cr.yaml", cr)
+	if len(nt.NonRootRepos) > 0 {
```
Comment (sdowell) [inline e2e/nomostest/config_sync.go:878]: refactoring this `if` statement makes this diff unnecessarily large imo. also makes the `if` block quite long which makes it harder to follow

File: e2e/nomostest/nt.go (hunk 2 above)
Comment (karlkfi) [reply e2e/nomostest/nt.go:442]: I had a brain wave so I'm refactoring these to `nt.expectedObjects` and `ExpectedRootSyncObjectCount` and adding Add & Remove methods.

File: e2e/nomostest/config_sync.go (hunk 3 above)
Comment (karlkfi) [reply e2e/nomostest/config_sync.go:879]: Previously `if len(nt.NonRootRepos) > 0 {` was returning before this block. So the behavior hasn't changed.

File: e2e/nomostest/config_sync.go (hunk 4 above)
Comment (karlkfi) [reply e2e/nomostest/config_sync.go:878]: That's why i had removed the if statement in a previous PR, but then that broke metric validation, which was being done after the If block.

File: e2e/nomostest/config_sync.go (hunk 3 above)
Comment (sdowell) [reply e2e/nomostest/config_sync.go:879]: Ah I see, my mistake

File: e2e/nomostest/config_sync.go (hunk 4 above)
Comment (karlkfi) [reply e2e/nomostest/config_sync.go:878]: hypothetically, but it's possible they may diverge in the future. This is just some bits extracted from https://github.com/GoogleContainerTools/kpt-config-sync/pull/412, which changes all the metrics validation impls. I just wanted them to be consistent first.

File: e2e/nomostest/nt.go (hunk 2 above)
Comment (sdowell) [reply e2e/nomostest/nt.go:442]: ack, let me know when it's ready for review

File: e2e/nomostest/config_sync.go (hunk 4 above)
Comment (karlkfi) [reply e2e/nomostest/config_sync.go]: done
